tree/tree.go

```diff
@@ -0,0 +1,723 @@
+package tree
+
+import (
+	"fmt"
+	"os"
+
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/charmbracelet/lipgloss"
+	ltree "github.com/charmbracelet/lipgloss/tree"
+
+	"github.com/charmbracelet/bubbles/help"
+	"github.com/charmbracelet/bubbles/key"
+)
+
+// StyleFunc allows the tree to be styled per item.
+type StyleFunc func(children Nodes, i int) lipgloss.Style
+
+// Styles contains style definitions for this tree component. By default, these
+// values are generated by DefaultStyles.
+type Styles struct {
+	SelectionCursor       lipgloss.Style
+	HelpStyle             lipgloss.Style
+	TreeStyle             lipgloss.Style
+	selectedNodeFunc      StyleFunc
+	SelectedNodeStyle     lipgloss.Style
+	SelectedNodeStyleFunc StyleFunc
+	nodeFunc              StyleFunc
+	NodeStyle             lipgloss.Style
+	NodeStyleFunc         StyleFunc
+}
+
+// DefaultStyles returns a set of default style definitions for this tree
+// component.
+func DefaultStyles() (s Styles) {
+	s.NodeStyle = lipgloss.NewStyle()
+	s.SelectedNodeStyle = lipgloss.NewStyle().
+		Background(lipgloss.Color("62")).
+		Foreground(lipgloss.Color("18")).
+		Bold(true)
+	s.nodeFunc = func(_ Nodes, _ int) lipgloss.Style {
+		return s.NodeStyle
+	}
+	s.selectedNodeFunc = func(_ Nodes, _ int) lipgloss.Style {
+		return s.SelectedNodeStyle
+	}
+	s.SelectionCursor = lipgloss.NewStyle().Foreground(lipgloss.Color("9"))
```

Comment:
currently this does nothing, but we could use it to show a pointing cursor. Is that something we want?